packages/sanity/src/core/store/_legacy/datastores.ts

```diff
@@ -134,11 +134,12 @@ export function useDocumentStore(): DocumentStore {
   const workspace = useWorkspace()
 
   const serverActionsEnabled = useMemo(() => {
-    // If it's explicitly disabled, we'll just return a stream that emits `false`
-    return workspace.serverActions?.enabled === false
-      ? of(false)
+    const configFlag = workspace.__internal_serverDocumentActions?.enabled
+    // If it's explicitly set, let it override the feature toggle
+    return typeof workspace.__internal_serverDocumentActions?.enabled === 'boolean'
```

Comment:
Nit: This should probably use `configFlag` which is bound to the value and used below.

```suggestion
    return configFlag === 'boolean'
```

Reply:
Thanks! Fixed